### PR TITLE
track forum and proposal events in the public api

### PR DESCRIPTION
**Pull request by mattcasey**

### CODE REVIEW

File: lib/forums/posts/voteForumComment.ts [hunk 1]
```diff
@@ -41,17 +41,13 @@ export async function voteForumComment({
...
-    if (category) {
-      trackUserAction(upvoted ? 'upvote_comment' : 'downvote_comment', {
-        resourceId: commentId,
-        spaceId: post.spaceId,
-        userId,
-        categoryName: category.name,
-        postId: post.id
-      });
-    }
+    trackUserAction(upvoted ? 'upvote_comment' : 'downvote_comment', {
```
Comment (motechFR) [inline lib/forums/posts/voteForumComment.ts:44]: Edge case: I think we can have null, which removes the upvote, but doesn't downvote

File: pages/api/v1/forum/comments/[commentId]/index.ts [hunk 2]
```diff
@@ -29,8 +27,15 @@ handler.delete(deletePostComment).put(requireKeys(['contentMarkdown'], 'body'),
...
  *
  */
-async function deletePostComment(req: NextApiRequest, res: NextApiResponse) {
+
+const handler = superApiHandler();
+handler.delete(deletePostComment).put(requireKeys(['contentMarkdown'], 'body'), updatePostCommentEndpoint);
+
+async function updatePostCommentEndpoint(req: NextApiRequest, res: NextApiResponse) {
   // This should never be undefined, but adding this safeguard for future proofing
+  if (!req.query.commentId) {
```
Comment (motechFR) [inline pages/api/v1/forum/comments/[commentId]/index.ts:36]: We could have this as additional middleware requireKeys(['commentId'], 'query')

File: pages/api/v1/forum/comments/[commentId]/index.ts [hunk 3]
```diff
@@ -44,17 +49,12 @@ async function deletePostComment(req: NextApiRequest, res: NextApiResponse) {
...
   });
 
-  if (result) {
-    await prisma.postComment.delete({
-      where: {
-        id: result.id
-      }
-    });
-  }
+  await deletePostComment({ commentId: result.id, userId: result.createdBy });
```
Comment (motechFR) [inline pages/api/v1/forum/comments/[commentId]/index.ts:57]: Are we getting userId from API? If so we might want to pass that here instead

File: pages/api/v1/proposals/[proposalId]/comments/[commentId]/vote.ts [hunk 4]
```diff
@@ -76,7 +78,8 @@ async function voteComment(req: NextApiRequest, res: NextApiResponse<PublicApiPr
       }
     },
     select: {
-      id: true
+      id: true,
```
Comment (motechFR) [inline pages/api/v1/proposals/[proposalId]/comments/[commentId]/vote.ts:81]: Suggestion; We can refactor this logic like you did with forum posts endpoint

**Review by motechFR — APPROVED** [review]

File: lib/forums/posts/voteForumComment.ts (hunk 1 above)
Comment (mattcasey) [reply lib/forums/posts/voteForumComment.ts:44]: thanks, you are right, i think this was an existing bug

File: pages/api/v1/forum/comments/[commentId]/index.ts (hunk 3 above)
Comment (mattcasey) [reply pages/api/v1/forum/comments/[commentId]/index.ts:57]: no, we dont ask for userId here

**Pull request merged**